server/controller/chore.js

```diff
@@ -196,3 +220,55 @@ export const deleteChore = async (req, res) => {
     res.status(500).json({ message: "Internal Server Error" });
   }
 }; //deleteChore
+
+/*
+ ** This function checks off a chore by updating its nextOccurrence and checkedOff params
+ ** @param {Object} req - The request object
+ ** @param {Object} res - The response object
+ */
+ export const checkOffChore = async (req, res) => {
+  try {
+    const id = req.params.id;
+
+    const chore = await Chore.findById({ _id: id });
+
+    if (chore === null) {
+      return res
+        .status(404)
+        .json({ message: `Chore with id ${id} was not found` });
+    }
+
+    // mapping interval into days
+    var intervalToDays;
+    if (chore.frequency.interval == "days") {
+        intervalToDays = 1;
+    } else if (chore.frequency.interval == "weeks") {
+        intervalToDays = 7;
+    } else if (chore.frequency.interval == "months") {
+        intervalToDays = 30; // just defaulting to 30
+    } else {
+        intervalToDays = 365; // interval is in years
+    }
+
+    // calculating how many days this chore is to be repeated
+    const repeatInDays = chore.frequency.quantity * intervalToDays;
+    // console.log(`Repeat in days: ${repeatInDays}`)
+
+    checkOffTime = Date.now();
+
+    // adding the check off time in the array of chores last checked off list
+    chore.lastCheckedOff.push(checkOffTime);
+
+    // updating next occurrence
+    const msInDay = 24 * 60 * 60 * 1000; // number of milliseconds in a day
+    const nextOccurrence = checkOffTime + (repeatInDays * msInDay); // all time to be stored in milliseconds
```

Comment:
I think this should be a separate function as well. Something like `getNextOccurrence(fromDate)`.